mypy/meet.py

```diff
@@ -243,6 +245,10 @@ def visit_tuple_type(self, t: TupleType) -> Type:
                 items.append(self.meet(t.items[i], self.s.items[i]))
             # TODO: What if the fallbacks are different?
             return TupleType(items, t.fallback)
+        # Special case: all Tuple[t1, t2, ...] are subtypes of plain tuple.
+        elif (isinstance(self.s, Instance) and self.s.type.fullname() == 'builtins.tuple'
```

Comment:
We could also handle cases like the meet of `Tuple[int, int]` and `Tuple[int, ...]`. However, as this is slightly trickier, it's okay to add a comment here about this and create a follow-up issue. Which would you prefer?

Implementing the general case shouldn't be hard, though it's not urgent really. The meet of arbitrary `Tuple[t1, <...>,  tn]` and `Tuple[s, ...]` (where `<...>` would not be a Python ellipsis but a mathematical one) would be `Tuple[meet(t1, s), ..., meet(tn, s)]`, I think. So the meet of types `Tuple[X, Y]` and `Tuple[Z, ...]` would be `Tuple[meet(X, Z), meet(Y, Z)]`, etc.